virtcontainers/clh.go

```diff
@@ -373,8 +375,29 @@ func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
 }
 
 func (clh *cloudHypervisor) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
-	clh.Logger().WithField("function", "hotplugAddDevice").Warn("hotplug add device not supported")
-	return nil, nil
+	span, _ := clh.trace("hotplugAddDevice")
+	defer span.Finish()
+
+	switch devType {
+	case vfioDev:
+		device := devInfo.(*config.VFIODev)
+
+		cl := clh.client()
+		ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+		defer cancel()
+
+		_, _, err := cl.VmmPingGet(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = cl.VmAddDevicePut(ctx, chclient.VmAddDevice{Path: device.SysfsDev})
+
+		return nil, openAPIClientError(err)
+	default:
+		return nil, fmt.Errorf("cannot hotplug device: unsupported device type '%v'", devType)
+	}
+
```

Comment:
Nit: you may want to separate the whole `case vfioDev:` block a standalone function that may be easier to maintain and ready

Reply:
Agree! moved to its own function.